src/verify/verify.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2022 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { UrlString } from "@inrupt/solid-client";
+import {
+  VerifiableCredential,
+  getVerifiableCredentialApiConfiguration,
+  isVerifiableCredential,
+} from "../common/common";
+import fallbackFetch from "../fetcher";
+
+export default async function isValidVc(
+  vc: VerifiableCredential | URL | UrlString,
+  options: Partial<{
+    fetch?: typeof fetch;
+    verificationEndpoint?: UrlString;
+  }> = {}
+): Promise<{ checks: string[]; warnings: string[]; errors: string[] }> {
+  const fetcher = options.fetch ?? fallbackFetch;
+
+  let vcObject;
+  // This test passes for both URL and UrlString
+  if (vc.toString().startsWith("http")) {
+    // vc is either an IRI-shaped string or a URL object. In both
+    // cases, vc.toString() is an IRI.
+    const vcResponse = await fetcher(vc.toString());
+    vcObject = await vcResponse.json();
+  } else {
+    vcObject = vc;
+  }
+  if (!isVerifiableCredential(vcObject)) {
+    throw new Error(
+      `The request to [${vc}] returned an unexpected response: ${JSON.stringify(
+        vcObject,
+        null,
+        "  "
+      )}`
+    );
```

Comment:
Could errors be extracted to an error folder and typed?
Something like `src/error/invalidVcError.ts`:
```typescript
export class VerifiableCredentialTypeError extends Error {
  constructor(vc: string, vcObject: string) {
    super(
      `The request to [${vc}] did not return a verifiable credential.\nActual: ${vcObject}\nExpected: A verifiable credential JSON object`
    );
  }
}
```

Reply:
I'd rather use our errors library and do this in one clean sweep, so I'll do this in a different PR.